src/compiler/scala/tools/nsc/symtab/SymbolLoaders.scala

```diff
@@ -309,7 +313,7 @@ abstract class SymbolLoaders {
       // errors. More concretely, the classfile parser calls "sym.companionModule", which calls
       // "isModuleNotMethod" on the companion. After refchecks, this method forces the info, which
       // may run the classfile parser. This produces the error.
-      enteringPhase(phaseBeforeRefchecks)(classfileParser.parse(classfile, root))
+      enteringPhase(phaseBeforeRefchecks)(classfileParser.parse(classfile, clazz, module))
```

Comment:
Hmm.. I think it's a bit risky to remove the phase travel here, the classfile parser and unpickler call various methods on symbols. I'll update the comment.